site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -512,7 +513,12 @@ public function showNewAnnouncement() {
     private function sendSocketMessage($msg_array) {
         $msg_array['user_id'] = $this->core->getUser()->getId();
         $msg_array['page'] = $this->core->getConfig()->getCourse() . "-office_hours_queue";
-        $client = new Client($this->core);
-        $client->send($msg_array);
+        try {
+            $client = new Client($this->core);
+            $client->send($msg_array);
+        }
+        catch (WebSocket\ConnectionException $e) {
+            $this->core->addNoticeMessage("WebSocket Server is down, Page won't load dynamically.");
+        }
     }
```

Comment:
Can you please add documentations above this function and also type hints.

Additionally can you change the error message to be "WebSocket Server is down, page won't load dynamically."  The `p` shouldn't be capitalized.

Reply:
Is the documentation added here clear enough?